learning_resources_search/views.py

```diff
@@ -74,11 +86,97 @@ def get(self, request):
             return Response(errors, status=400)
 
 
+class UserSearchSubscriptionViewSet(viewsets.ViewSet):
+    """
+    View for listing percolate query subscriptions for a user
+    """
+
+    permission_classes = (IsAuthenticated,)
+    serializer_class = UserPercolateQueryRequestSerializer
+    filter_backends = (DjangoFilterBackend,)
```

Comment:
Suggestion:

```python
class UserSearchSubscriptionViewSet(viewsets.ModelViewSet):
    """
    View for listing percolate query subscriptions for a user
    """

    permission_classes = (IsAuthenticated,)
    serializer_class = PercolateQuerySerializer
    filter_backends = (DjangoFilterBackend,)
    http_method_names = ["get"]

    def get_queryset(self):
        """
        Generate a QuerySet for fetching valid PercolateQueries for this user

        Returns:
            QuerySet of PercolateQuery objects subscribed to by request user
        """
        queryset = self.request.user.percolate_queries.all()
        for backend in list(self.filter_backends):
            queryset = backend().filter_queryset(self.request, queryset, view=self)
        return queryset
```

Then you can remove the `list` and `filter_queryset` functions below.

Reply:
much cleaner. done